### PR TITLE
Hammerhead.upload doesn't work properly (close #193)

### LINKED ISSUE #193: Hammerhead.upload doesn't work properly
Hammerhead.upload published [here](https://github.com/superroma/testcafe-hammerhead/blob/master/src/client/index.js#L40):
```js
this.upload = this.sandbox.upload.upload;
```
but it should be:
```js
this.upload = this.sandbox.upload.upload.bind(this.sandbox.upload);
```
to have the correct context in the `UploadSandbox.upload` function

### PR DESCRIPTION
**Pull request by churkin**
\cc @inikulin @AlexanderMoskovkin

### CODE REVIEW

File: test/client/fixtures/sandbox/upload-test.js [hunk 1]
```diff
@@ -361,12 +361,12 @@ module('server errs');
...
     var input = $('<input type="file">')[0];
 
-    uploadSandbox.upload(input, './err_file.txt')
+    uploadSandbox.doUpload(input, './err_file.txt')
         .then(function (errs) {
             strictEqual(errs.length, 1);
             strictEqual(errs[0].err, 34);
 
-            uploadSandbox.upload(input, ['./err_file1.txt', './file.txt', './err_file2.txt'])
+            uploadSandbox.doUpload(input, ['./err_file1.txt', './file.txt', './err_file2.txt'])
```
Comment (inikulin) [inline test/client/fixtures/sandbox/upload-test.js]: Can you get rid of nested `then`s in tests, please?

**Comment by inikulin** [comment]
PR lgtm, but it would be nice to fix tests nested promises

**Comment by churkin** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by inikulin** [comment]
lgtm

**Pull request merged**